src/dataExplorer/components/Results.tsx

```diff
@@ -144,58 +124,157 @@ const Results: FC = () => {
+  const {result, status} = useContext(ChildResultsContext)
+  const {range} = useContext(PersistanceContext)
+
+  return (
+    <div className="data-explorer-results--view">
+      <View
+        loading={status}
+        properties={view.properties}
+        result={result?.parsed}
+        timeRange={range}
+        hideTimer
+      />
+    </div>
+  )
+}
+
+const WrappedOptions: FC = () => {
+  // use parent `results` so all metadata is present for the viz options
+  const {result} = useContext(ResultsContext)
+  const {setResult, setStatus} = useContext(ChildResultsContext)
+  const {view, setView /* setViewOptions*/} = useContext(ResultsViewContext)
+
+  const updateChildResults = useCallback(
+    update => {
+      setView({
+        ...view,
+        properties: {
+          ...view.properties,
+          ...update,
+        },
+      })
+    },
+    [setStatus, setResult]
+  )
+
+  // TODO: make component, for QxBuilder-specific graph subquery options
+  const subQueryOptions = null
+
+  return (
+    <>
+      {subQueryOptions}
+      {/* <SubQueryOptions update={setViewOptions} /> */}
+      <ViewOptions
+        properties={view.properties}
+        results={result.parsed}
+        update={updateChildResults}
+      />
+    </>
+  )
+}
+
+const GraphHeader: FC = () => {
+  const {view, setView} = useContext(ResultsViewContext)
+  const {result} = useContext(ResultsContext)
+  const {result: subQueryResult} = useContext(ChildResultsContext)
+  const {launch} = useContext(SidebarContext)
+
+  const launcher = () => {
+    launch(<WrappedOptions />)
   }
 
-  const dataExists = res && Object.entries(res).length
   const updateType = viewType => {
     setView({
       state: 'graph',
       properties: SUPPORTED_VISUALIZATIONS[viewType].initial,
     })
   }
 
-  const launcher = () => {
-    launch(<WrappedOptions />)
+  const dataExists = !!result?.parsed
+  const subqueryReturnsData = !!subQueryResult?.parsed
+  let titleText = 'Configure Visualization'
+  if (!dataExists) {
+    titleText = 'No data to visualize yet'
+  }
+  if (!subqueryReturnsData) {
+    titleText = 'Graph customization options returned no data'
+  }
+
+  return (
+    <>
+      <ViewTypeDropdown
+        viewType={view.properties.type}
+        onUpdateType={updateType}
+      />
+      <Button
+        text="Customize"
+        icon={IconFont.CogSolid_New}
+        onClick={launcher}
+        status={dataExists ? ComponentStatus.Default : ComponentStatus.Disabled}
+        color={ComponentColor.Default}
+        titleText={titleText}
+        className="de-config-visualization-button"
+      />
+    </>
+  )
+}
+
+const Results: FC = () => {
+  const [search, setSearch] = useState('')
+  const {status} = useContext(ResultsContext)
+  const {view, setView} = useContext(ResultsViewContext)
+
+  let resultView
+
+  if (status === RemoteDataState.NotStarted) {
+    resultView = <EmptyResults />
+  } else {
+    if (view.state === 'table') {
```

Comment:
This is a good candidate to make into a TypeScript [enum](https://www.typescriptlang.org/docs/handbook/enums.html) so we don't have to do raw string comparisons.

Reply:
yaz. I like where your head's at.